### PR TITLE
bugfix to allow weightless autograd.nn.AutoModule

### PR DESCRIPTION
**Pull request by szagoruyko**
support was incomplete.

### CODE REVIEW

**Comment by alexbw** [comment]
Thanks for the pull. Was there a test that was failing previously? If not, could you please add one so that we can be robust against a regression?

File: src/auto/AutoModule.lua [hunk 1]
```diff
@@ -55,11 +55,13 @@ return function(moduleName)
       end
 
       function module:accGradParameters(input, gradOutput, scale)
-         if not self.grads then
-            self.grads = self.b({input=input, weight=self.weight, bias=self.bias}, gradOutput)
+         if self.weight then
```
Comment (fmassa) [inline src/auto/AutoModule.lua]: I think you need to be more careful with that, there is at least one module in nn which has bias but no weight, see https://github.com/torch/nn/blob/master/Add.lua#L9

Comment (alexbw) [inline src/auto/AutoModule.lua]: For the PR, a test that tests the full set of combos ((weight, bias),
(weight), (bias), ()) would be best.

Comment (szagoruyko) [inline src/auto/AutoModule.lua]: Without this even creating a simple transfer AutoModule fails, so I'd rather merge this and add tests in another PR. Will take a look when I have time, just started using autograd.

**Comment by alexbw** [comment]
Sounds good.

**Pull request merged**